src/resolve/resolved-link.c

```diff
@@ -301,6 +301,27 @@ static int link_update_dns_servers(Link *l) {
         return r;
 }
 
+static int link_update_default_route(Link *l) {
+        int r;
+
+        assert(l);
+
+        r = sd_network_link_get_dns_default_route(l->ifindex);
+        if (r == -ENODATA) {
+                r = 0;
+                goto clear;
+        }
+        if (r < 0)
+                goto clear;
+
+        l->default_route = r > 0;
+        return 0;
+
+clear:
+        l->default_route = -1;
+        return r;
```

Comment:
Hmm, resetting the value even in the error path is unexpected.

Reply:
it's how all the other link_update_xyz() calls operate too: if we can't read the data from networkd we'll unset the options... i.e. parse errors invalidate data, they don't mean old data remains. And I'd argue it's nice to do it that way since it means we expose the same behaviour if resolved is just started up and reads networkd data for the first time, and if it is already running and the data is corrupted while it is operating, since in both cases it will now invalidate the data. If we'd leave the old data in place then the behaviour would differ between those two cases, if you follow what i mean...

Reply:
OK, makes sense.